### PR TITLE
Added bilateral filter

### LINKED ISSUE #54: Add "bilateral filter" feature in filters section -
Implement bilateral filter and incorporate that code in filters.py

### PR DESCRIPTION
**Pull request by noobkid2411**
# Description:

Added bilateral filter code to filters.py

Fixes #54 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

 
# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

# Screenshots / Video:
 I have taken two blurred images from geeksforgeeks and geekscoders for better visualisation of the results and a third one I've added from my own collection to show how bilateral filter evens the image out,if there are any other pictures you'd like to see it work on, I'll add more.
1)
Before filter:
taj
After filter:
bilateraltaj

2)
Before filter:
woman
After filter:
bilateralwoman

3)
Before filter:
me

After filter:
bilateralme

### CODE REVIEW

File: flow2ml/Filters.py [hunk 1]
```diff
@@ -131,6 +131,29 @@ def applygaussian(self,classPath):
         
         # saving the image.
         plt.imsave(classPath+"/GaussianImages/gaussian"+image, cv2.cvtColor(gaussian, cv2.COLOR_RGB2BGR))
+        
+        
+        def applybilateral(self,classPath):
```
Comment (chebroluharika) [inline flow2ml/Filters.py:136]: Align this line properly

File: flow2ml/Filters.py [hunk 2]
```diff
@@ -131,6 +131,29 @@ def applygaussian(self,classPath):
...
+    ''' 
+      Applies Bilateral Filter to all the images in the given folder. 
+      Args : 
+        classPath : (string) directory containing images for a particular class.
+    '''
+    try:
+      os.mkdir(classPath+"/Bilaterally_FilteredImages")    
+    except:
+      pass
```
Comment (chebroluharika) [inline flow2ml/Filters.py:145]: Add proper exception handling message here

**Comment by chebroluharika** [comment]
@noobkid2411 please attach screenshots showing the application of filter. And also add documentation in Readme for this filter

File: flow2ml/Filters.py (hunk 2 above)
Comment (noobkid2411) [reply flow2ml/Filters.py:145]: could you help me a little on this one,what exacly do i need to add?

File: flow2ml/Filters.py (hunk 1 above)
Comment (noobkid2411) [reply flow2ml/Filters.py]: done

**Comment by noobkid2411** [comment]
@chebroluharika added screenshots of application and added bilateral filter to the readme file as well.

**Comment by chebroluharika** [comment]
@noobkid2411 also add this filter name in sample code section in Readme

File: flow2ml/Filters.py (hunk 2 above)
Comment (chebroluharika) [reply flow2ml/Filters.py:145]: Keep like this:
except Exception as e:
            print(f"Cannot create directory with given name due to {e}")

File: flow2ml/Filters.py [hunk 3]
```diff
@@ -131,6 +131,31 @@ def applygaussian(self,classPath):
...
+      Applies Bilateral Filter to all the images in the given folder. 
+      Args : 
+        classPath : (string) directory containing images for a particular class.
+    '''
+    try:
+      os.mkdir(classPath+"/Bilaterally_FilteredImages")    
+    
+    except Exception as e:
+      print(f"Cannot create directory with given name due to: {e}")
+      pass
```
Comment (chebroluharika) [inline flow2ml/Filters.py:147]: Remove “pass” here

**Pull request merged**